packages/react-events/src/dom/Input.js

```diff
@@ -0,0 +1,323 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {
+  ReactDOMEventResponder,
+  ReactDOMResponderEvent,
+  ReactDOMResponderContext,
+} from 'shared/ReactDOMTypes';
+
+import React from 'react';
+import {DiscreteEvent} from 'shared/ReactTypes';
+
+type InputEventType = 'change' | 'beforechange' | 'valuechange';
+type KeyboardEventType = 'keydown' | 'keyup' | 'keypress';
+
+type InputResponderProps = {
+  disabled: boolean,
+  onKeyDown: (e: InputEvent) => void,
+  onKeyUp: (e: InputEvent) => void,
+  onKeyPress: (e: InputEvent) => void,
+  onBeforeChange: (e: InputEvent) => void,
+  onChange: (e: InputEvent) => void,
+  onValueChange: (value: string | boolean) => void,
+};
+
+type InputEvent = {|
+  data: string,
+  isComposing: boolean,
+  inputType: boolean,
```

Comment:
Why inputType is boolean? It should be a string, isn't it? Also looks like it's not supported in IE and Edge
https://developer.mozilla.org/en-US/docs/Web/API/InputEvent/inputType

Reply:
That was a typo. We aren't looking to polyfill this though, if it exists we pass it on.